AWSPinpointTests/AWSPinpointNotificationManagerTests.m

```diff
@@ -0,0 +1,153 @@
+    };
+
+    _journeyPushPayload = @{
+        @"data": @{
+            @"pinpoint": @{
+                @"journey": @{
+                    @"journey_id": @"testJourneyId"
+                }
+            }
+        }
+    };
+}
+
+- (void)tearDown {
+    [self.pinpoint.analyticsClient.eventRecorder removeAllEvents];
+    [self.pinpoint.analyticsClient removeAllGlobalFeatureAttributes];
+    [super tearDown];
+}
+
+- (void)testAddPinpointMetadataForCampaignEvent {
+    AWSPinpointEvent *pushNotificationEvent = [self.pinpoint.analyticsClient createEventWithEventType:@"_campaign.received_foreground"];
+    [_pinpoint.notificationManager addPinpointMetadataForEvent:pushNotificationEvent
+                                              withNotification:self.campaignPushPayload];
+    XCTAssertTrue([pushNotificationEvent.eventType isEqualToString:@"_campaign.received_foreground"]);
+    XCTAssertTrue([pushNotificationEvent.allAttributes[@"campaign_id"] isEqualToString:@"testCampaignId"]);
+}
+
+- (void)testAddPinpointMetadataForJourneyEvent {
+    AWSPinpointEvent *pushNotificationEvent = [self.pinpoint.analyticsClient createEventWithEventType:@"_journey.received_background"];
+    [_pinpoint.notificationManager addPinpointMetadataForEvent:pushNotificationEvent
+                                              withNotification:self.journeyPushPayload];
+    XCTAssertTrue([pushNotificationEvent.eventType isEqualToString:@"_journey.received_background"]);
+    XCTAssertTrue([pushNotificationEvent.allAttributes[@"journey_id"] isEqualToString:@"testJourneyId"]);
+}
+
+- (void)testAddPinpointMetadataForUnknownEvent {
+    NSDictionary *userInfo = @{
+        @"data": @{
+            @"pinpoint": @{
+                @"randomKey": @{
+                    @"randomAttribute": @"randomValue"
+                }
+            }
+        }
+    };
+    AWSPinpointEvent *pushNotificationEvent = [self.pinpoint.analyticsClient createEventWithEventType:@"_unknown.received_background"];
+    [_pinpoint.notificationManager addPinpointMetadataForEvent:pushNotificationEvent
+                                              withNotification:userInfo];
+    XCTAssertEqual([pushNotificationEvent.allAttributes count], 0);
+}
+
+- (void)testAddGlobalPinpointMetadataForCampaignNotification {
+    [_pinpoint.notificationManager addGlobalPinpointMetadataForNotification:self.campaignPushPayload];
+    XCTAssertEqual([self.pinpoint.analyticsClient.globalFeatureAttributes count], 1);
+    XCTAssertEqual(self.pinpoint.analyticsClient.globalFeatureAttributes[@"campaign_id"], @"testCampaignId");
+}
+
+- (void)testAddGlobalPinpointMetadataForJourneyNotification {
+    [_pinpoint.notificationManager addGlobalPinpointMetadataForNotification:self.journeyPushPayload];
+    XCTAssertEqual([self.pinpoint.analyticsClient.globalFeatureAttributes count], 1);
+    XCTAssertEqual(self.pinpoint.analyticsClient.globalFeatureAttributes[@"journey_id"], @"testJourneyId");
+}
+
+- (void)testBuildEventType {
```

Comment:
These should be broken up into separate, individual, isolated unit tests

Reply:
Will update.